libraries/AP_IBus_Telem/AP_IBus_Telem.cpp

```diff
@@ -0,0 +1,674 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+   i-BUS telemetry for FlySky/Turnigy receivers and other peripherals
+   (eg iA6B, iA10) by Nicole Ashley <nicole@ashley.kiwi>.
+
+   Originally based on work by Jan Verhulst:
+     https://github.com/ArduPilot/ardupilot/pull/16545
+
+   Libraries used for reference and inspiration:
+
+   * iBUStelemetry
+     https://github.com/Hrastovc/iBUStelemetry
+
+   * IBusBM
+     https://github.com/bmellink/IBusBM
+
+   * BetaFlight
+     https://github.com/betaflight/betaflight/blob/master/src/main/telemetry/ibus_shared.c
+ */
+
+#include <AP_IBus_Telem/AP_IBus_Telem.h>
+
+#if AP_IBUS_TELEM_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Vehicle/AP_Vehicle.h>
+#include <AP_BattMonitor/AP_BattMonitor.h>
+#include <AP_Arming/AP_Arming.h>
+#include <AP_GPS/AP_GPS.h>
+#include <AP_RPM/AP_RPM.h>
+
+// All the sensors we can accurately provide are listed here.
+// i-BUS will generally only query up to 15 sensors, so subjectively
+// higher-value sensors are sorted to the top to make the most of a
+// small telemetry window. In the future these could be configurable.
+static const AP_IBus_Telem::SensorDefinition _sensors[] = {
```

Comment:
```suggestion
static const AP_IBus_Telem::SensorDefinition sensors[] {
```

It doesn't matter here where it's a static-const, but omitting the equals can have benefits - not constructing objects on the stack and increased type-checking.  So just as a pattern we're preferring it.

... also removed the leading underscore on the variable name.  We're moving away from prepending underscores to things - so if we could remove them from the object's variables that would be good...

Reply:
I originally had no underscores but when I did a style guide check I noticed it said to put them in, so I did 😆 

I'll update!

Reply:
Oh, wow.  Wiki PR coming up for DevCall to fix that, my apologies.

Reply:
https://github.com/ArduPilot/ardupilot_wiki/pull/6150

Reply:
Wonderful, thank you :)